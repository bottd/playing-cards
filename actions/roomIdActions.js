export const actionTypes = {
  resetRoomID: 'RESET_ROOM_ID',
  updateRoomID: 'UPDATE_ROOM_ID',
}

export const resetRoomID = () => ({
  type: actionTypes.resetRoomID,
});

export const updateRoomID = id => ({
  type: actionTypes.updateRoomID,
  id,
});
