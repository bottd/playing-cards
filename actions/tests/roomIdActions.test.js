import { actionTypes, resetRoomID, updateRoomID } from '../roomIdActions';

describe('roomIdActions', () => {
  describe('action types', () => {
    it('Should match snapshot', () => {
      expect(actionTypes).toMatchSnapshot();
    });
  });

  describe('resetRoomID', () => {
    it('Should return an action with correct type', () => {
      const expected = { type: 'RESET_ROOM_ID' };
      const result = resetRoomID();
      expect(result).toEqual(expected);
    });
  });

  describe('updateRoomID', () => {
    it('Should return an action with correct type', () => {
      const expected = 'UPDATE_ROOM_ID';
      const result = updateRoomID(1);
      expect(result.type).toBe(expected);
    });
    it('Should return an action with room id', () => {
      const expected = 1;
      const result = updateRoomID(expected);
      expect(result.id).toBe(expected);
    });
    it('Should match snapshot', () => {
      const result = updateRoomID(1);
      expect(result).toMatchSnapshot();
    });
  });
});
