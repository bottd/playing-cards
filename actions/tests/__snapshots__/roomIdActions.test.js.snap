// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`roomIdActions action types Should match snapshot 1`] = `
Object {
  "resetRoomID": "RESET_ROOM_ID",
  "updateRoomID": "UPDATE_ROOM_ID",
}
`;

exports[`roomIdActions updateRoomID Should match snapshot 1`] = `
Object {
  "id": 1,
  "type": "UPDATE_ROOM_ID",
}
`;
